pycloudlib/cloud.py

```diff
@@ -62,13 +66,39 @@ def __init__(
         self._check_and_set_config(config_file, required_values)
 
         user = getpass.getuser()
+        # check if id_rsa or id_ed25519 keys exist in the user's .ssh directory
```

Comment:
This turns `__init__()` mostly into a key-handling function. I would prefer a helper method `_get_ssh_keys()`[1] to contain this logic and get called from `__init__()`

[1] maybe a better name, I didn't put any thought into that one

Reply:
Haha agreed! Way too much going on there. And would allow for actually unit testing that (if so desired)